File: agregation/java/Joueur.java
package packAgre;

public class Joueur {
    private String nom;
    private String prenom;
    private int numero;
    private int age;
    private String poste;

    //constructeur par initialisation
    public Joueur(String nom, String prenom, int numero, int age, String poste){
        this.nom=nom;
        this.prenom=prenom;
        this.numero=numero;
        this.age=age;
        this.poste=poste;
    }

    //getters
    public String getNom(){
        return nom;
    }

    public String getPrenom(){
        return prenom;
    }

    public int getNumero(){
        return numero;
    }

    public int getAge(){
        return age;
    }

    public String getPoste(){
        return poste;
    }

    //pas de setters pour l'exemple car non utile

    //pas de méthode car pas necessaire dans l'exemple

    //destructeur
    public void finalize(){
        System.out.println("Le joueur n'existe plus");
    }
}

File: agregation/cpp/Joueur.cpp
#include "Joueur.h"

//constructeur par initialisation
Joueur::Joueur(string nom, string prenom, int numero, int age, string poste):nom(nom),prenom(prenom),numero(numero),age(age),poste(poste){};
//destructeur
Joueur::~Joueur(){
    cout<<"Desctructeur joueur"<<endl;
}

//getters
string Joueur::getPrenom() const{
    return prenom;
}

string Joueur::getNom() const{
    return nom;
}

int Joueur::getNumero() const{
    return numero;
}

int Joueur::getAge() const{
    return age;
}

string Joueur::getPoste() const{
    return poste;
}